app/uk/gov/hmrc/timetopayproxy/connectors/TtpConnector.scala

```diff
@@ -95,34 +87,22 @@ class DefaultTtpConnector @Inject()(appConfig: AppConfig, httpClient: HttpClient
                   implicit ec: ExecutionContext,
                   hc: HeaderCarrier
                 ): TtppEnvelope[UpdatePlanResponse] = {
-    val path = if(appConfig.useIf) "individuals/time-to-pay/quote" else "debts/time-to-pay/quote"
+    val path = if (appConfig.useIf) "individuals/time-to-pay/quote" else "debts/time-to-pay/quote"
     val url = s"${appConfig.ttpBaseUrl}/$path/${updatePlanRequest.customerReference.value}/${updatePlanRequest.planId.value}"
 
-    val response: Future[Either[ConnectorError, UpdatePlanResponse]] =
+    EitherT(
       httpClient
-        .PUT[UpdatePlanRequest, UpdatePlanResponse](url, updatePlanRequest)
-        .map(r => r.asRight[ConnectorError])
-        .recover {
-          case e: HttpException => ConnectorError(e.responseCode, e.message).asLeft[UpdatePlanResponse]
-          case e: UpstreamErrorResponse => ConnectorError(e.statusCode, e.getMessage()).asLeft[UpdatePlanResponse]
-        }
-
-    TtppEnvelope(response)
+        .PUT[UpdatePlanRequest, Either[TtppError, UpdatePlanResponse]](url, updatePlanRequest)
```

Comment:
Shouldn't TtppError be TimeToPayError, that you created?

Reply:
As TtppError is the error on the other side of the Api, the format that Pega expects.